services/api/src/resources/problem/resolvers.ts

```diff
@@ -9,17 +9,37 @@ const logger = require('../../logger');
 export const getAllProblems: ResolverFn = async (
   root,
   args,
-  { sqlClient, hasPermission }
+  {
+    sqlClient,
+    hasPermission,
+    models,
+    keycloakGrant
+  },
 ) => {
   let rows = [];
+  let userProjectIds = [];
 
   try {
-    if (!R.isEmpty(args)) {
-      rows = await problemHelpers(sqlClient).getAllProblems(args.source, args.environment, args.envType, args.severity);
-    }
-    else {
-      rows = await query(sqlClient, Sql.selectAllProblems({source: [], environmentId: 0, environmentType: [], severity: []}));
+    await hasPermission('problem', 'viewAll');
+
+    // Get all projects
+    const prep = prepare(sqlClient, `SELECT id FROM project`);
+    let projects  = await query(sqlClient, prep(args));
+    userProjectIds = projects.map(p => p.id);
+  }
+  catch (err) {
+    if (!keycloakGrant) {
+      logger.warn('No grant available for getAllProblems');
+      return [];
     }
+
+    userProjectIds = await models.UserModel.getAllProjectsIdsForUser({
+      id: keycloakGrant.access_token.content.sub,
+    });
```

Comment:
After we get these project ids, let's filter out any projects where the following test fails

```
      await hasPermission('problem', 'view', {
          project: project,
      });
```